src/tally.F90

```diff
@@ -845,6 +1032,122 @@ subroutine score_fission_eout(p, t, i_score)
   end subroutine score_fission_eout
 
 !===============================================================================
+! SCORE_FISSION_DELAYED_EOUT handles a special case where we need to store
+! delayed neutron production rate with an outgoing energy filter (think of a
+! fission matrix). In this case, we may need to score to multiple bins if there
+! were multiple neutrons produced with different energies.
+!===============================================================================
+
+  subroutine score_fission_delayed_eout(p, t, i_score)
+
+    type(Particle), intent(in)       :: p
+    type(TallyObject), intent(inout) :: t
+    integer, intent(in)              :: i_score ! index for score
+
+    integer :: i             ! index of outgoing energy filter
+    integer :: j             ! index of delayedgroup filter
+    integer :: d             ! delayed group
+    integer :: g             ! another delayed group
+    integer :: d_bin         ! delayed group bin index
+    integer :: n             ! number of energies on filter
+    integer :: k             ! loop index for bank sites
+    integer :: bin_energyout ! original outgoing energy bin
+    real(8) :: score         ! actual score
+    real(8) :: E_out         ! energy of fission bank site
+    logical :: d_found = .false. ! bool to inidicate if delayed group was found
+
+    ! save original outgoing energy and delayed group bins
+    i = t % find_filter(FILTER_ENERGYOUT)
+    j = t % find_filter(FILTER_DELAYEDGROUP)
+    bin_energyout = matching_bins(i)
+
+    ! Get number of energies on filter
+    n = size(t % filters(i) % real_bins)
+
+    ! Since the creation of fission sites is weighted such that it is
+    ! expected to create n_particles sites, we need to multiply the
+    ! score by keff to get the true nu-fission rate. Otherwise, the sum
+    ! of all nu-fission rates would be ~1.0.
+
+    ! loop over number of particles banked
+    do k = 1, p % n_bank
+
+      ! get the delayed group
+      g = fission_bank(n_bank - p % n_bank + k) % delayed_group
+      d_found = .FALSE.
+
+      ! check if the particle was born delayed
+      if (g /= 0) then
+
+        ! determine score based on bank site weight and keff
+        score = keff * fission_bank(n_bank - p % n_bank + k) % wgt
+
+        ! determine outgoing energy from fission bank
+        E_out = fission_bank(n_bank - p % n_bank + k) % E
+
+        ! check if outgoing energy is within specified range on filter
+        if (E_out < t % filters(i) % real_bins(1) .or. &
+             E_out > t % filters(i) % real_bins(n)) cycle
+
+        ! check if delayed group is in delayed group bins
+        if (j > 0) then
```

Comment:
This subroutine will be entered when `delayed-nu-fission` is requested with an energy out filter. The delayed group filter (with index `j` in this subroutine) does not necessarily have to be present. If it is not present, the total `delayed-nu-fission` rate is tallied; If it is present; the `delayed-nu-fission` rate is tallied for the requested delayed group bins.

I will push changes shortly that modify this subroutine to clarify the procedure for the cases where a delayed group filter is present and where a delayed group filter is not present.